cmd/buildclient/main.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2022, Sylabs, Inc. All rights reserved.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	"github.com/sylabs/scs-build-client/internal/app/buildclient"
+)
+
+const (
+	keyAccessToken      = "auth-token"
+	keySkipTLSVerify    = "skip-verify"
+	keyArch             = "arch"
+	keyArtifactFileName = "output"
+	keyFrontendURL      = "url"
+	keyForceOverwrite   = "force"
+	keyImageSpec        = "image-spec"
+)
+
+var Usage = func() {
+	fmt.Fprintf(os.Stderr, `
+Usage: buildclient [opts] <path to build definition>
+
+    Build and push artifact to cloud library:
+
+        buildclient --image-spec library://user/project/image:tag alpine.def
+
+    Build, push artfifact to cloud library and download locally:
+
+        buildclient --image-spec library://user/project/image:tag --output /tmp/alpine_3.sif alpine.def
+
+    Build and push ephemeral artifact to cloud library:
+
+        buildclient alpine.def
+
+    Note: ephemeral artifacts are short-lived and are usually deleted within 24 hours.
+
+Options:
+`)
+	pflag.PrintDefaults()
+}
+
+func getFlagSet() (*pflag.FlagSet, error) {
+	fs := pflag.CommandLine
+
+	fs.String(keyAccessToken, "", "Access token")
+	fs.Bool(keySkipTLSVerify, false, "Skip SSL/TLS certificate verification")
+	fs.String(keyArch, buildclient.DefaultBuildArch, "Requested build architecture")
```

Comment:
Wondering if it might be better to use the system architecture here, rather than default to `amd64`? For current CI systems, this will in practice be `amd64`. But it's possible someone will run this outside of a CI system, and in that case the least surprising behaviour is likely the current architecture?

```suggestion
	fs.String(keyArch, runtime.GOARCH, "Requested build architecture")
```